### PR TITLE
Add get cpu credit metric

### PR DESCRIPTION
**Pull request by shoukoo**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -14,6 +15,82 @@ import (
...
+	}
+
+	defer rep.Body.Close()
+	body, err := ioutil.ReadAll(rep.Body)
+	if err != nil {
+		return "", err
+	}
+
+	instanceId := string(body)
+	return instanceId, nil
```
Comment (alexlance) [inline main.go]: I'd condense these two lines (ie no point defining a variable, when it's just used once, briefly)

File: main.go [hunk 2]
```diff
@@ -70,6 +147,37 @@ func getSwap() int {
...
 
+func getCredit() int {
+
+	instanceId, err := getInstanceId()
+
+	if err != nil {
+		return 0
+	}
+
+	commandStr := "aws cloudwatch get-metric-statistics --namespace AWS/EC2 --metric-name CPUCreditBalance --region ap-southeast-2 --dimensions Name=InstanceId,Value=" + instanceId + " --start-time $(date -d '10 minute ago' +%s) --end-time=$(date +%s) --period 3600 --statistics Minimum --unit Count | jq '.Datapoints[0].Minimum'"
```
Comment (alexlance) [inline main.go]: I think start time should just be 1 minute ago? (ie this monitor runs once per minute)

File: main.go [hunk 3]
```diff
@@ -70,6 +147,37 @@ func getSwap() int {
...
+		return 0
+	}
+
+	instance, err := getInstanceDetails()
+	if err != nil {
+		return 0
+	}
+	percentageUsage := (v / instance.MaximumCredits) * 100
+
+	return int(percentageUsage)
```
Comment (alexlance) [inline main.go]: Please condense the last two lines again, no need for a variable which just exists momentarily - even if its name does help explain what it does, so too does the function name.

File: main.go [hunk 4]
```diff
@@ -96,12 +204,13 @@ func main() {
               "Component/CPU[percent]": %d,
               "Component/Disk[percent]": %d,
               "Component/Memory[percent]": %d,
-              "Component/Swap[percent]": %d
+			  "Component/Swap[percent]": %d,
+			  "Component/CPUCredit[percent]": %d
```
Comment (alexlance) [inline main.go]: Extra whitespace/tab.

**Review by alexlance — COMMENTED** [review]
I think the newrelic-monitor binary file has been included in your PR. Did you use git commit -A ? :)

File: main.go [hunk 5]
```diff
@@ -14,6 +15,82 @@ import (
 
 const Version = "v0.4"
```
Comment (alexlance) [inline main.go]: Please bump the version number.

File: main.go [hunk 6]
```diff
@@ -14,6 +15,82 @@ import (
...
 
+type EC2InstanceType struct {
+	Name                 string
```
Comment (alexlance) [inline main.go]: We may as well omit the Name property, as it isn't used, and is just duplicated data.

File: main.go [hunk 7]
```diff
@@ -14,6 +15,82 @@ import (
...
+	if err != nil {
+		return "", err
+	}
+
+	instanceId := string(body)
+	return instanceId, nil
+}
+
+func getInstanceDetails() (EC2InstanceType, error) {
+
```
Comment (alexlance) [inline main.go]: Nuke newline

File: main.go [hunk 8]
```diff
@@ -14,6 +15,82 @@ import (
...
+	defer rep.Body.Close()
+	body, err := ioutil.ReadAll(rep.Body)
+	if err != nil {
+		return instance, err
+	}
+
+	instanceName := string(body)
+	instance = instanceDetails[instanceName]
+
+	return instance, nil
```
Comment (alexlance) [inline main.go]: Condense last two line please

File: main.go [hunk 9]
```diff
@@ -70,6 +147,37 @@ func getSwap() int {
 	return int(v)
 }
 
+func getCredit() int {
+
```
Comment (alexlance) [inline main.go]: Nuke newline

File: main.go (hunk 4 above)
Comment (alexlance) [inline main.go]: And CPUCredit can just be named Credit - too.Thankyou

File: main.go [hunk 10]
```diff
@@ -14,6 +15,82 @@ import (
...
+			Name:                 "t3.nano",
+			MaximumCredits:       144,
+			CreditsEarnedPerHour: 6,
+		},
+		"t3.micro": EC2InstanceType{
+			Name:                 "t3.micro",
+			MaximumCredits:       288,
+			CreditsEarnedPerHour: 12,
+		},
+	}
```
Comment (alexlance) [inline main.go:98]: There are 7 t2 and 7 t3 instance types, we should add them all

File: main.go (hunk 2 above)
Comment (shoukoo) [reply main.go]: in the doc , it says `CPU credit metrics are available at a five-minute frequency only.`
I'll change to 5.

**Pull request merged**